sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -250,15 +250,15 @@ static void register_device_with_provisioning_service(void)
   uint8_t mqtt_payload[MQTT_PAYLOAD_BUFFER_LENGTH];
   size_t mqtt_payload_length;
 
-  AZ_PUSH_IGNORE_DEPRECATIONS
+  _az_PUSH_IGNORE_DEPRECATIONS
```

Comment:
Should these not be available for customers should they want to ignore the deprecations? Or are we saying they should find their own mechanism to (if they want) since they are risking using deprecated functions?

Reply:
> Or are we saying they should find their own mechanism to (if they want) since they are risking using deprecated functions?

That.

Using a deprecated function is not supported, so we shouldn't make it convenient and easy to circumvent that warning to discourage use. If the user wants to do so, they can use whatever mechanism makes sense to them.

Reply:
> Using a deprecated function is not supported

I don't think that's correct. V1 must have support for all deprecated functions for the entire lifetime if we are to be backward compatible.

> so we shouldn't make it convenient and easy to circumvent that warning

This is debatable: if an existing application is already using the function, it is safe for them to continue to do so until they will be forced to use something else in V2 or they require additional support / functionality offered by the new, non-deprecated function.

/cc @RickWinter , @JeffreyRichter

Reply:
Azure SDKs can make breaking changes we just strongly discourage it due to the customer pain.
I haven't been involved in the deprecation macro stuff and so I'm not sure what its purpose/goal is. 
Seems to me that there isn't much need to deprecate anything - we just ship a new version that is (publicly) missing the thing we don't want anymore, and notify customers of the break and how to best migrate from the old thing to the new thing.